test/gtest/reducecalculation.hpp

```diff
@@ -186,15 +195,30 @@ struct ReduceCalculationTest : public ::testing::TestWithParam<ReduceCalculation
             }
         }
 
-        output = tensor<T>{out_dims};
-        std::fill(output.begin(), output.end(), std::numeric_limits<T>::quiet_NaN());
+        if(isLogicalCalculation)
+        {
+            logical_output     = tensor<uint8_t>{out_dims};
+            logical_ref_output = tensor<uint8_t>{out_dims};
 
-        ref_output = tensor<T>{out_dims};
-        std::fill(ref_output.begin(), ref_output.end(), std::numeric_limits<T>::quiet_NaN());
+            std::fill(logical_output.begin(), logical_output.end(), 0);
```

Comment:
in general, you should fill output received results from GPU with `quite_NaN` as in some cases, it would help you determine errors caused by data transfer

Reply:
solved